odoo/addons/base/migrations/11.0.1.3/pre-migration.py

```diff
@@ -45,6 +47,107 @@ def fill_cron_action_server_pre(env):
     )
 
 
+def ensure_country_state_id_on_existing_records(cr):
+    """Suppose you have country states introduced manually.
+    This method ensure you don't have problems later in the migration when
+    loading the res.country.state.csv"""
+    with open(get_module_resource('base', 'res', 'res.country.state.csv'),
+              newline='') as country_states_file:
+        states = csv.reader(country_states_file, delimiter=',', quotechar='"')
+        for row, state in enumerate(states):
+            if row == 0:
+                continue
+            data_name = state[0]
+            country_code = state[1]
+            name = state[2]
+            state_code = state[3]
+            # first: query to ensure the existing odoo countries have
+            # the code of the csv file, because maybe some code has changed
+            cr.execute(
```

Comment:
This is not needed, as the pre-requisite is always to have a DB of the previous version totally updated, and this data is noupdate="0" (being a CSV), so no possible problem. If any problem, go to first step ;)